added vs code instructions to the installation guide

jc-berger opened the pull request:
This PR isn't made for a particular issue, but rather, it's intended to make a downstream version of the upstream doc in [PR #3977](https://github.com/openshift/odo/pull/3977). 

There's no specific version this PR and doc change is intended for. 

@openshift/team-documentation, please do a peer review at your convenience. 

Please note, this PR must be tested by somebody from the odo QE team.

jc-berger commented:
@openshift/team-documentation, I'm following up regarding the review of the squashed commit. This PR doesn't need to be merged until October 29th, so no rush at all. Thanks!

prietyc123 commented:
LGTM

cdrage commented:
LGTM / approve by me!

/lgtm

boczkowska commented:
@jc-berger since this is a procedure, it should be in a separate module. See https://redhat-documentation.github.io/modular-docs/

Preeticp commented on cli_reference/developer_cli_odo/installing-odo.adoc:
```diff
@@ -23,3 +23,17 @@ include::modules/developer-cli-odo-installing-odo-on-linux-on-ibm-z.adoc[levelof
 
 include::modules/developer-cli-odo-installing-odo-on-macos.adoc[leveloffset=+1]
+
+= Installing odo in VS Code
```
As Yana mentioned the instructions in this section should be in a separate procedure module.
Once you create the procedure module you add it in this file as an include, for example see line 23 and 24 above. 
This is an Assembly file, where we add only the includes for the files/sections we want under this assembly. Kind of like a chapter with multiple sections.

Preeticp commented on cli_reference/developer_cli_odo/installing-odo.adoc:
```diff
@@ -23,3 +23,17 @@ include::modules/developer-cli-odo-installing-odo-on-linux-on-ibm-z.adoc[levelof
+
+== Plugin installation
+
+. Launch VS Code Quick Open (Ctrl+P)
```
Make Quick Open bold
Change Ctrl+P to `Ctrl+P`

Preeticp commented on cli_reference/developer_cli_odo/installing-odo.adoc:
```diff
@@ -23,3 +23,17 @@ include::modules/developer-cli-odo-installing-odo-on-linux-on-ibm-z.adoc[levelof
+
+. Launch VS Code Quick Open (Ctrl+P)
+
+. Paste the following command:
```
Suggest changing Paste to Enter

jc-berger replied:
Done.

jc-berger commented:
@Preeticp and @boczkowska, thank you both for the review. I've added a procedure model. If it looks good to you two, I believe this PR will have passed all the rounds of review and will be ready to merge.

Preeticp commented:
/lgtm

jc-berger commented:
@peer-review-squad, would you please do a review? If it looks good to you, then this PR can be merged. Thanks!

bergerhoffer commented on cli_reference/developer_cli_odo/installing-odo.adoc:
```diff
@@ -5,7 +5,7 @@ include::modules/common-attributes.adoc[]
 toc::[]
 
-The following section describes how to install `{odo-title}` on different platforms using the CLI.
+The following section describes how to install `{odo-title}` on different platforms using the CLI as well as the IDE, Visual Studio Code (VS Code).
```
Suggest "... using the CLI or the Visual Studio Code (VS Code) IDE."

bergerhoffer commented on modules/developer-cli-odo-installing-odo-on-vs-code.adoc:
```diff
@@ -0,0 +1,19 @@
+//
+// * cli_reference/developer_cli_odo/installing-odo.adoc
+
+[id="installing-odo-on-vs-code"]
```
You're missing the "_{context}" in this anchor. Should be:

```
[id="installing-odo-on-vs-code_{context}"]
```

bergerhoffer commented on modules/developer-cli-odo-installing-odo-on-vs-code.adoc:
```diff
@@ -0,0 +1,19 @@
+. Enter the following command:
+
+----
+  $ ext install redhat.vscode-openshift-connector
```
Remove the extra indentation at the beginning of the command.

bergerhoffer commented on modules/developer-cli-odo-installing-odo-on-vs-code.adoc:
```diff
@@ -0,0 +1,19 @@
+
+= Installing odo on VS Code
+
+The https://marketplace.visualstudio.com/items?itemName=redhat.vscode-openshift-connector[OpenShift VS Code extension] uses both `odo` and the `oc` binary to interact with your Kubernetes or OpenShift cluster. Install odo on VS Code to work with these features.
```
Suggest removing "Kubernetes or". Because these are specifically OpenShift docs.

Actually I would also replace "OpenShift" with {product-title}, because I think these docs show for OKD. So:

```
... to interact with your {product-title} cluster.
```

bergerhoffer replied:
> Install odo on VS Code to work with these features.

Are you installing odo on VS code with this procedure, or should you say that you are installing this OpenShift VS Code extension on VS Code?

bergerhoffer commented on modules/developer-cli-odo-installing-odo-on-vs-code.adoc:
```diff
@@ -0,0 +1,19 @@
+
+The https://marketplace.visualstudio.com/items?itemName=redhat.vscode-openshift-connector[OpenShift VS Code extension] uses both `odo` and the `oc` binary to interact with your Kubernetes or OpenShift cluster. Install odo on VS Code to work with these features.
+
+== Plugin installation
```
Instead of this heading, this should be a simple Procedure heading (to comply with mod docs):

```
.Procedure
```

bergerhoffer replied:
This still needs to be addressed.

bergerhoffer commented on modules/developer-cli-odo-installing-odo-on-vs-code.adoc:
```diff
@@ -0,0 +1,19 @@
+
+== Plugin installation
+
+. Launch VS Code Quick Open with `Ctrl+P`
```
I assume that these steps already assume that you have VS Code open? I'd suggest adding a step saying so with "Launch VS Code." or "Open VS Code" or whatever is appropriate.

bergerhoffer replied:
You're also missing a period here.

bergerhoffer commented on modules/developer-cli-odo-installing-odo-on-vs-code.adoc:
```diff
@@ -0,0 +1,19 @@
+= Installing odo on VS Code
+
+The https://marketplace.visualstudio.com/items?itemName=redhat.vscode-openshift-connector[OpenShift VS Code extension] uses both `odo` and the `oc` binary to interact with your Kubernetes or OpenShift cluster. Install odo on VS Code to work with these features.
+
```
This procedure also seems to assume that you already have VS code installed. So I would add that as a prerequisite:

```
.Prerequisites

* You have installed VS Code.
```

bergerhoffer commented on modules/developer-cli-odo-installing-odo-on-vs-code.adoc:
```diff
@@ -0,0 +1,19 @@
+. Launch VS Code Quick Open with `Ctrl+P`
+
+. Enter the following command:
+
```
You'll need a `+` on this line for the following code block to be properly indented with this step.

bergerhoffer reviewed (commented):
Let me know if you have any questions on these comments!

bergerhoffer commented on modules/developer-cli-odo-installing-odo-on-vs-code.adoc:
```diff
@@ -0,0 +1,19 @@
+
+== Plugin installation
+
+. Launch VS Code Quick Open with `Ctrl+P`
```
Also, the "+" here shouldn't be in the backticks. Should be:

```
`Ctrl`+`P`
```

bergerhoffer reviewed (commented):
One more thing I noticed.

jc-berger commented:
@bergerhoffer, thanks, Andrea, for the review. I've addressed the comments and resolved the conversations. Would you please confirm all looks good to merge?

bergerhoffer commented on modules/developer-cli-odo-installing-odo-on-vs-code.adoc:
```diff
@@ -0,0 +1,25 @@
+
+The https://marketplace.visualstudio.com/items?itemName=redhat.vscode-openshift-connector[OpenShift VS Code extension] uses both `odo` and the `oc` binary to interact with your {product title} cluster. Install the OpenShift VS Code extension on VS Code.
+
+== Prerequisites
```
This needs to be `.Prerequisites`. See the mod docs template file here for details: https://raw.githubusercontent.com/redhat-documentation/modular-docs/master/modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc

jc-berger replied:
Okay, would that mean all the other sections, need a `.` instead of a `=` or `==`?

bergerhoffer replied:
".Procedure" and ".Prerequisites" are special headings that are part of the mod docs template for a procedure module. Same for if you were to have  ".Verification steps" or ".Additional resources" in the module.

Now if you're in an assembly, you would use "== Prerequisites" and "== Additional resources".

Otherwise, if you just have any other random heading outside of these ones that are part of the template, then yeah normally you would use the == Heading format.

bergerhoffer commented on modules/developer-cli-odo-installing-odo-on-vs-code.adoc:
```diff
@@ -0,0 +1,25 @@
+
+= Installing odo on VS Code
+
+The https://marketplace.visualstudio.com/items?itemName=redhat.vscode-openshift-connector[OpenShift VS Code extension] uses both `odo` and the `oc` binary to interact with your {product title} cluster. Install the OpenShift VS Code extension on VS Code.
```
You're missing the dash in the attribute, so it's not being properly replaced. Should be `{product-title}`.

bergerhoffer replied:
And the last sentence now sounds a little abrupt without the "to work with these features." Recommend adding that, or something like it, back in.

jc-berger replied:
Okay, thanks! 

Funny that the previous comment wasn't addressed...I thought I had. Thanks for catching!

bergerhoffer commented on modules/developer-cli-odo-installing-odo-on-vs-code.adoc:
```diff
@@ -0,0 +1,25 @@
+
+== Prerequisites
+
+. You have installed VS Code.
```
Prerequisites should be an unnumbered bullet (`*`)

bergerhoffer reviewed (commented):
Just a few more things!

bergerhoffer commented on modules/developer-cli-odo-installing-odo-on-vs-code.adoc:
```diff
@@ -0,0 +1,25 @@
+
+* You have installed VS Code.
+
+.Plugin installation
```
This literally has to be called `.Procedure`, not `.Plugin installation`.

bergerhoffer reviewed (commented):
Just one thing, but should be the last of it!

bergerhoffer commented:
LGTM now, thanks!

bergerhoffer commented:
/cherrypick enterprise-4.7

bergerhoffer commented:
/cherrypick enterprise-4.6
